src/main/scala/org/broadinstitute/dsde/firecloud/webservice/WorkspaceApiService.scala

```diff
@@ -110,8 +110,21 @@ trait WorkspaceApiService extends HttpService with FireCloudRequestBuilding
             }
           } ~
           path("methodconfigs") {
-            requireUserInfo() { _ =>
-              passthrough(workspacePath + "/methodconfigs", HttpMethods.GET, HttpMethods.POST)
+            get {
+              requireUserInfo() { _ =>
+                passthrough(workspacePath + "/methodconfigs", HttpMethods.GET)
+              }
+            } ~
+            post {
+              requireUserInfo() { _ =>
+                entity(as[MethodConfiguration]) { methodConfig =>
+                  requestContext =>
+                    if (methodConfig.outputs.collect { case param if param._2.value.startsWith("this.library:") => param }.isEmpty)
```

Comment:
Also probably not for this PR ... still thinking out loud. This feels like it is more a utility or library support kind of check and seems like too much logic to be putting inside of a route.